src/nomad_analysis/schema.py

```diff
@@ -201,13 +202,19 @@ def set_jupyter_notebook_name(
             archive (EntryArchive): The archive containing the section.
             logger (BoundLogger): A structlog logger.
         """
-        file_name = (
-            self.name.replace(' ', '_')
-            + f'_{self.analysis_type.lower()}_notebook.ipynb'
-        )
-        if not archive.m_context.raw_path_exists(self.notebook):
+        if self.name:
+            file_name = (
```

Comment:
```python
filename = f'{self.name.replace(' ', '_')}_{self.analysis_type.lower()}_notebook.ipynb}'
```

Reply:
Thanks for the suggestion, but I think both the ways of writing are equivalent

Reply:
They are, but it is a bit complicated to read at first sight.

please, either go with summing up strings, or with `f'...'`, don't mix it.

Reply:
Ah I see, then I will accept the suggestion :smile:

Reply:
Furthermore, what if `self.analysis_type` is None?

Reply:
the value is hard-coded in the child classes, so it shouldn't be None during the runtime

Reply:
I think that is not reason enough: if someone creates a new class from this one, they should now that a priori they need to define `analysis_type` or the logic of this abstract class will break.

I will: either add error handling here, or add a `default='Generic'` to the Quantity (which might be better)

Reply:
I agree, I already added `default` for this ;)